packages/itwin/grouping-mapping-widget/src/widget/components/CustomCalculationTable.tsx

```diff
@@ -6,30 +6,50 @@ import {
   SvgDelete,
   SvgEdit,
   SvgMore,
+  SvgStatusWarning,
 } from "@itwin/itwinui-icons-react";
 import {
+  DefaultCell,
   DropdownMenu,
   IconButton,
   MenuItem,
 } from "@itwin/itwinui-react";
-import React, { useCallback } from "react";
-import type { CellProps } from "react-table";
-import type { CustomCalculation } from "@itwin/insights-client";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import type { CellProps, CellRendererProps } from "react-table";
+import type { CalculatedProperty, CalculatedPropertyLinks, CustomCalculation, DataType, GroupProperty, QuantityType } from "@itwin/insights-client";
 import { useMappingClient } from "./context/MappingClientContext";
-import { PropertyNameCell } from "./PropertyNameCell";
 import { PropertyTable } from "./PropertyTable";
 import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import { usePropertiesContext } from "./context/PropertiesContext";
+import { handleError } from "./utils";
+import { getOutliers } from "./PropertyValidationUtils";
+
+interface CustomCalculationTableData extends CustomCalculation {
+  id: string;
+  propertyName: string;
+  formula: string;
+  dataType: DataType;
+  quantityType: QuantityType;
+  _links: CalculatedPropertyLinks;
```

Comment:
Why include these properties when you are extending `CustomCalculation`?

Reply:
This was from when it was being used as a class. Now that it is using interface, I have updated this.